### DISCUSSION TITLE
ebus adapter shield pins and interface

### DISCUSSION THREAD

**Discussion by tgrosz** [john30/ebusd#1299] [Q&A]
Hi, can you help me where to start if I'd like to use the shield's pins? Is there documentation somewhere? I'd like to control a relay and use the pins. I saw the rest api - does it have docs and is there any other interface?

**Comment by ptm26** [comment]
From:
https://community.home-assistant.io/t/an-ebusd-add-on/344852/410

----------------------------------------------------------------------------
I have been working on the same problem myself and I just found a solution.

Complete lack of documentation on this feature but I got it working using the REPL API. In the Easi> web interface under configuration, make sure httpd is set to no TLS, no authorization for the following commands to work.

```
#Set Pin 0 High
curl -X PUT http://192.168.1.136/api/v1/repl -d "io -h 0"

#Set Pin 0 Low
curl -X PUT http://192.168.1.136/api/v1/repl -d "io -l 0"
```
Then translate that into your Home Assistant configuration.yaml file:

```
shell_command:
  central_heating_on:  curl -X PUT http://192.168.1.136/api/v1/repl -d "io -h 0"
  central_heating_off: curl -X PUT http://192.168.1.136/api/v1/repl -d "io -l 0"
```

Depending on your relay type you made need Low for ON, so adjust the commands accordingly. I hope that helps!

----------------------------------------------------------

**Reply by daclerk** [reply]
Thanks a lot ptm26 for this info! I was waiting all the time for MQTT support for the IO pins in the firmware. But now I can for the time being continue with your solution. I will test it as soon as I have some spare time…
